src/lean_spec/subspecs/containers/state/state.py

```diff
@@ -86,23 +90,26 @@ def generate_genesis(cls, genesis_time: Uint64, num_validators: Uint64) -> "Stat
         )
 
         # Build the genesis block header for the state.
+        genesis_body = BlockBody(attestations=Attestations(data=[]))
+
         genesis_header = BlockHeader(
             slot=Slot(0),
             proposer_index=ValidatorIndex(0),
             parent_root=Bytes32.zero(),
             state_root=Bytes32.zero(),
-            body_root=hash_tree_root(BlockBody(attestations=Attestations(data=[]))),
+            body_root=hash_tree_root(genesis_body),
         )
 
         # Assemble and return the full genesis state.
         return cls(
             config=genesis_config,
             slot=Slot(0),
             latest_block_header=genesis_header,
-            latest_justified=Checkpoint(root=Bytes32.zero(), slot=Slot(0)),
-            latest_finalized=Checkpoint(root=Bytes32.zero(), slot=Slot(0)),
+            latest_justified=Checkpoint.default(),
+            latest_finalized=Checkpoint.default(),
             historical_block_hashes=HistoricalBlockHashes(data=[]),
             justified_slots=JustifiedSlots(data=[]),
+            validators=Validators(data=[]),
```

Comment:
Should we add validators import function into the specs? Could go as a separate PR though.

Reply:
Specifying how to parse validator data from a file